### PR TITLE
Add a thermal species ( ncdm )

### PR DESCRIPTION
**Pull request by rainwoodman**
This PR adds a thermal species that carries coherent thermal velocities.

The thermal species is assumed to be matter-like, and made of massive-1/2 spin particles that follows fermi-dirac distribution.

At most 3 sub-species are supported.

The corherent thermal velocities follows the scheme from Arka Banerjee.

TODO list

- [ ] Verify the large scale growth of ncdm at low redshift against linear theory; using box sizes of 1 Gpc/h, comparable with Arka's original paper.

- [ ] Find a way to smooth out the initial grid structure; target ncdm spatial resolution is 8 Mpc/h per pre-split particle.

### CODE REVIEW

File: src/lua-runtime-fastpm.lua [hunk 1]
```diff
@@ -33,11 +33,17 @@ function schema.output_redshifts.action(output_redshifts)
...
 end
 
-schema.declare{name='omega_m',           type='number', required=true, default=0.3 }
+schema.declare{name='omega_m',           type='number', required=true, default=0.3, help='cdm + baryon density parameter at z=0'}
+schema.declare{name='omega_ncdm',           type='number', required=false, default=0.0, help='ncdm density parameter at z=0'}
 schema.declare{name='h',                 type='number', required=true, default=0.7, help="Dimensionless Hubble parameter"}
 schema.declare{name='pm_nc_factor',      type='array:number',  required=true, help="A list of {a, PM resolution}, "}
 schema.declare{name='np_alloc_factor',   type='number', required=true, help="Over allocation factor for load imbalance" }
 schema.declare{name='compute_potential',   type='boolean', required=false, default=false, help="Calculate the gravitional potential."}
+schema.declare{name='n_ncdm',   type='number', required=false, default=1, help="Number of ncdm particles."}
```
Comment (rainwoodman) [inline src/lua-runtime-fastpm.lua]: There is a way to get the number of items in `m_ncdm`, thus you don't need a separate `n_ncdm` parameter. Take a look how this is obtained for the `time_step` parameter.

Comment (adrianbayer) [reply src/lua-runtime-fastpm.lua]: Done 0a7c3f69b489486a7bae72bb2e250039cad51f1b

**Comment by rainwoodman** [comment]
@adrianbayer I will merge this to the master branch as it contains several useful functionality fixes that can be used in regular CDM runs. We can continue looking into the neutrino behavior after this.

**Comment by rainwoodman** [comment]
@adrianbayer I have merged the current branch as is to the master.

We shall open a new PR for new code changes. You plan to add the initial ncdm clustering?

**Comment by adrianbayer** [comment]
@rainwoodman Yes, I think it would be good. I'll begin on it tomorrow!

**Pull request merged**